### PR TITLE
gigasecond: add test template

### LINKED ISSUE #1950: gigasecond: add test template
## Task
Create a test template file for `gigasecond`

### Details
1. If this is your first time contributing to this repository, welcome!
    - Contributing to a repository on GitHub requires a [fork](https://help.github.com/en/articles/fork-a-repo)
1. Create a branch on your local repository called `gigasecond-add-test-template`
    - Run: `git checkout -b gigasecond-add-test-template`
1. Clone problem-specifications repository; this is used to populate the template
    - Run: `git clone https://github.com/exercism/problem-specifications ../problem-specifications`
1. Create `exercises/gigasecond/.meta/template.j2`
    - See [Generator Documentation] for more information
1. Generate test file from new template
    - Run: `bin/generate_tests.py gigasecond`

At this point, if the generator fails, or you are otherwise stuck,
please feel free to reach out! It is perfectly acceptable to submit
an incomplete PR in order to get feedback or assistance.

### Resources
- [Contributing Guide]
- [Generator Documentation]
- [Closing issues using keywords]

[Contributing Guide]: https://github.com/exercism/python/blob/master/docs/CONTRIBUTING.md
[Generator Documentation]: https://github.com/exercism/python/blob/master/docs/GENERATOR.md
[Closing issues using keywords]: https://help.github.com/en/articles/closing-issues-using-keywords

### PR DESCRIPTION
**Pull request by simonbcodes**
Added the gigasecond test template. The test specification in `problem-specifications/exercises/gigasecond/canonical-data.json` seems to unnecessarily nest `cases` within `cases`, so I had to work around that, but I'm not entirely sure that is correct.

See #1950

### CODE REVIEW

File: exercises/gigasecond/.meta/template.j2 [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{%- import "generator_macros.j2" as macros with context -%}
+from datetime import datetime
+{{ macros.header() }}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases[0]["cases"] -%}
```
Comment (cmccandless) [inline exercises/gigasecond/.meta/template.j2]:
```suggestion
	{% for case in supercase["cases"] %}{% for case in cases -%}
```

File: exercises/gigasecond/.meta/template.j2 [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    def {{ case["description"] | to_snake }}(self):
+        test_value = "{{ case["input"]["moment"] }}"
+        expected = "{{ case["expected"] }}"
+        self.assertEqual({{ case["property"] }}(test_value), expected)
+    {% endfor %}
```
Comment (cmccandless) [inline exercises/gigasecond/.meta/template.j2]:
```suggestion
    {% endfor %}{% endfor %}
```

File: exercises/gigasecond/.meta/template.j2 [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{%- import "generator_macros.j2" as macros with context -%}
+from datetime import datetime
+{{ macros.header() }}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases[0]["cases"] -%}
+    def {{ case["description"] | to_snake }}(self):
+        test_value = "{{ case["input"]["moment"] }}"
+        expected = "{{ case["expected"] }}"
+        self.assertEqual({{ case["property"] }}(test_value), expected)
```
Comment (cmccandless) [inline exercises/gigasecond/.meta/template.j2]: I recommend the following instead, to minimize the diff on the test file:
```Jinja2
{%- set test_value = case["input"]["moment"] -%}
{%- set expected = case["expected"]
self.assertEqual(
    {{ case["property"] }}(
        datetime.strptime("{{ case["input"]["moment"] }}", "%Y-%m-%d")
    ),
    datetime.strptime("{{ case["expected"] }}", "%Y-%m-%dT%H:%M:%S")
)
```

File: exercises/gigasecond/.meta/template.j2 [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{%- import "generator_macros.j2" as macros with context -%}
+from datetime import datetime
```
Comment (cmccandless) [inline exercises/gigasecond/.meta/template.j2]:
```suggestion
from datetime import datetime, strptime
```

File: exercises/gigasecond/.meta/template.j2 [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+{{ macros.header() }}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases[0]["cases"] -%}
+    def {{ case["description"] | to_snake }}(self):
```
Comment (cmccandless) [inline exercises/gigasecond/.meta/template.j2]:
```suggestion
    def test_{{ case["description"] | to_snake }}(self):
```

**Comment by yawpitch** [comment]
@simonbcodes are you able to continue working on this one, with the various suggestions above?

**Comment by simonbcodes** [comment]
@yawpitch most likely not going to be able to continue with these changes, at least in the near future.

**Comment by yawpitch** [comment]
@simonbcodes ok, thanks ... I'm going to create a new branch on the repo, merge your work into that, and then continue from there. Thanks for the initial work.

**Pull request merged**